mapper/thin_edge_json/src/lib.rs

```diff
@@ -0,0 +1,574 @@
+//! A library to create the ThinEdgeJson from bytes of json data by Validating it.
+//! It also serialize the ThinEdgeJson data.
+
+use chrono::format::ParseError;
+use chrono::prelude::*;
+use json::JsonValue;
+use serde::{Deserialize, Serialize};
+
+/// ThinEdgeJson is represented in this struct
+/// Since json does not understand DateTime format, the time stamp is represented as a string
+/// Before populating the struct members the thinedge json values and names will be validated
+
+#[derive(Serialize, Deserialize)]
+pub struct ThinEdgeJson {
+    pub timestamp: String,
+    pub values: Vec<ThinEdgeValue>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum ThinEdgeValue {
+    Single(SingleValueMeasurement),
+    Multi(MultiValueMeasurement),
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SingleValueMeasurement {
+    pub name: String,
+    pub value: f64,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct MultiValueMeasurement {
+    pub name: String,
+    pub values: Vec<SingleValueMeasurement>,
+}
+
+impl ThinEdgeJson {
+    pub fn read_from_bytes(input: &[u8]) -> Result<Vec<u8>, ThinEdgeJsonError> {
+        Ok(ThinEdgeJson::from_utf8(input, Utc::now())?.serialize_thin_edge())
+    }
```

Comment:
What is the purpose of this function?

* At least badly named. When I see `ThinEdgeJson::read_form_bytes()`, I expect a function taking some bytes and returning a `ThinEdgeJson` struct not a different sequence of bytes.
* Seems to be used only for tests expecting an error. These tests can be re-written by directly calling `ThinEdgeJson::from_utf8()`.